services/leaderboard/src/main/java/io/openliberty/spacerover/leaderboard/rest/Leaderboard.java

```diff
@@ -136,56 +140,21 @@ private String getJsonListFromIterableDocuments(FindIterable<Document> docs) {
 	@Operation(summary = "Clear the leaderboard entries from the database.")
 	@DELETE
 	@Path("/")
-	public Response clear() {
+	public Response clear(@DefaultValue("") @QueryParam("playerID") String playerID) {
```

Comment:
when will deleting by player name be used? i'm worried since 2 players may coincidentally use the same name and we don't want to accidentally wipe them both.

Reply:
this will delete every player with the name, i could restrict it to delete the first one but .. how do we know that was the one intended to be deleted anyways? 

If we're worried about that we could always implement the player ID's as unique keys somehow, ensuring no one can use the same name twice?

Reply:
i think it might be a bit more useful right now to implement deleting one entry. we can use the `_id` property for this. this property is already included with each entry returned by the `GET /leaderboard/` endpoint.

i recommend separating the delete all and delete single resource into their own endpoints. [the mongo db guide has an example for deleteOne](https://github.com/OpenLiberty/guide-mongodb-intro/blob/prod/finish/src/main/java/io/openliberty/guides/application/CrewService.java#L260-L318).

i.e.,
delete many: `DELETE /leaderboard/`
delete one: `DELETE /leaderboard/{_id}`

Reply:
that makes sense, you don't like the idea of forcing uniqueness on the playerID?

Reply:
If we have dup names, then how do we choose which one to delete? Should we make it feasible in the UI to select which one to delete? something like a checkbox?

Reply:
Forcing uniqueness, might prevent players from playing multiple times? Would that overwrite the each score?

Reply:
i liked what we currently have for allowing players to repeat using their name (only downside is ppl might occasionally clash names, but is probably fine).

for the ui, i was going to add a delete button for each row in the leaderboard table

Reply:
I vote we have 3 options
DELETE on leaderboard: deletes everything
DELETE on leaderboard with playerID as a query parameter: deletes all entries with the playerID match
DELETE on leaderboard/id: deletes one record that matches the unique _id value.

Reply:
For jimmy's idea, he would have the _id which is already guaranteed unique for every row in our collection. So if the delete button is inline with the row he can supply the exact id for that entry.

Reply:
That makes sense. +1

Reply:
yeah i'm alright to keep the `DELETE on leaderboard with playerID as a query parameter` option. how about calling it playerName instead of playerID though?